.github/workflows/build_and_test.yml

```
@@ -6,45 +6,174 @@ on:
+      if: steps.check_commit_message.outputs.no_ci != 'true'
+      id: get_job_id
+      run: |
+        curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+        "https://api.github.com/repos/${{ github.repository }}/actions/runs/${{ github.run_id }}/jobs" \
+        | jq '.jobs[] | select(.name=="build") | .id' > job_id.txt
+        echo "job_id=$(cat job_id.txt)" >> $GITHUB_OUTPUT
+
+    - name: Determine Commit ID and Comitter Email
+      if: steps.check_commit_message.outputs.no_ci != 'true'
+      id: determine_commit_id_and_email_author
+      run: |
```

actionlint (1.7.7)

"github.event.head_commit.author.email" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)